core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -104,79 +110,67 @@ public ApplicationServer(T config, ThreadPool threadPool) {
     super.addBean(mbContainer);
 
     this.sslContextFactory = createSslContextFactory(config);
-    configureConnectors(sslContextFactory);
+    configureConnectors(sslContextFactory, namedListeners);
   }
 
   /**
    * TODO: delete deprecatedPort parameter when `PORT_CONFIG` is deprecated.
    * It's only used to support the deprecated configuration.
    */
   static List<URI> parseListeners(
-          List<String> listenersConfig,
+          List<String> listeners,
           int deprecatedPort,
           List<String> supportedSchemes,
-          String defaultScheme
-  ) {
+          String defaultScheme) {
```

Comment:
note: this method didn't really need to be modified, but at one point as i was implementing this it did, and i thought my changes represent an improvement, so left them.